### DISCUSSION TITLE
How do I use the `-check-config` option?

### DISCUSSION THREAD

**Discussion by msfjarvis** [glanceapp/glance#109] [Help] [answered]
I've written a small [NixOS](https://nixos.org) [module for Glance](https://github.com/msfjarvis/dotfiles/blob/ba1025689d6c99cb34932a2941048fb632c6f9ed/modules/nixos/glance/default.nix) for myself and was looking to add build-time validation for the config file but can't seem to figure out how to get it working.

When I run `glance -check-config true -config $CONFIG_FILE` it tries to make API calls for weather which is a bit surprising to me since I expected it to simply validate that the config can be parsed not that every field in it points to real and valid data on the Internet.

Just wanted to quickly check if this is intentional behavior or something you'd be open to changing.

**Comment by svilenmarkov** [comment, marked as answer]
Hey,

You are using it correctly, it's just that the weather widget is the only exception which makes an HTTP request during validation. It works that way due to Open Meteo's weather data endpoint requiring coordinates rather than location name which are retrieved once during the validation step. 

Multiple people have been unpleasantly surprised by this so I'll change it and make the retrieval of the coordinates happen during the widget update step instead, meaning that not having internet won't prevent the application from starting anymore. Expect to see the change in the next release and sorry for the confusion!
